app/src/main/java/com/edu/uni/augsburg/uniatron/ui/MainActivity.java

```diff
@@ -88,6 +99,26 @@ protected void onCreate(final Bundle savedInstanceState) {
         startService(new Intent(this, AppTrackingService.class));
     }
 
+    private void startOnboarding() {
+        final SharedPreferencesHandler sharedPrefsHandler =
+                MainApplication.getSharedPreferencesHandler(getApplicationContext());
+
+        // relaunch onboarding if permission request was ignored
+        if (sharedPrefsHandler.isFirstStart()
+                || needUsageAccessPermission()
+                || needBatteryWhitelistPermission()) {
+
+            startActivity(new Intent(this, OnboardingActivity.class));
```

Comment:
Maybe you should use the startActivityWithParentStack method in the MainActivity.